src/renderer/store/certificates/certificates.saga.ts

```diff
@@ -2,19 +2,36 @@ import { call, apply, all, takeEvery, put, select } from 'typed-redux-saga'
 import { PayloadAction } from '@reduxjs/toolkit'
 
 import { certificatesActions } from './certificates.reducer'
-import { createUserCsr, configCrypto } from '@zbayapp/identity'
+import { createUserCsr, configCrypto, CertFieldsTypes, parseCertificate, getCertFieldValue } from '@zbayapp/identity'
 import electronStore from '../../../shared/electronStore'
 import { actions as identityActions } from '../handlers/identity'
 import { registrationServiceAddress } from '../../../shared/static'
 import notificationsHandlers from '../../store/handlers/notifications'
 import { successNotification } from '../handlers/utils'
 import directMessagesSelectors from '../selectors/directMessages'
 
+const filterCertificates = (certificates: string[]): string[] => {
+  return certificates.filter((cert) => {
+    const parsedCert = parseCertificate(cert)
+
+    let isValid = true
+
+    for (const field of Object.keys(CertFieldsTypes)) {
+      if (cert && !getCertFieldValue(parsedCert, CertFieldsTypes[field])) {
+        isValid = false
+      }
+    }
+    return isValid
+  })
+}
+
 export function* responseGetCertificates(
   action: PayloadAction<ReturnType<typeof certificatesActions.responseGetCertificates>['payload']>
 ): Generator {
   const certificates = action.payload
-  yield* put(certificatesActions.setUsersCertificates(certificates.certificates))
+
+  const filteredCertificates = yield* call(filterCertificates, certificates.certificates)
```

Comment:
No need for yield call, there is no side effect in this function. 
you can call filterCertificates directly